pkg/specs/pgbouncer/services.go

```diff
@@ -18,39 +18,34 @@ package pgbouncer
 
 import (
 	corev1 "k8s.io/api/core/v1"
-	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 	"k8s.io/apimachinery/pkg/util/intstr"
 
 	apiv1 "github.com/cloudnative-pg/cloudnative-pg/api/v1"
 	pgBouncerConfig "github.com/cloudnative-pg/cloudnative-pg/pkg/management/pgbouncer/config"
+	"github.com/cloudnative-pg/cloudnative-pg/pkg/servicespec"
 	"github.com/cloudnative-pg/cloudnative-pg/pkg/utils"
 )
 
 // Service create the specification for the service of
 // pgbouncer
 func Service(pooler *apiv1.Pooler, cluster *apiv1.Cluster) *corev1.Service {
+	serviceTemplate := servicespec.NewFrom(pooler.Spec.ServiceTemplate).
+		WithName(pooler.Name).
+		WithNamespace(pooler.Namespace).
+		WithLabel(utils.PgbouncerNameLabel, pooler.Name).
+		WithLabel(utils.ClusterLabelName, cluster.Name).
+		WithServiceType(pooler.Spec.ServiceTemplate.Spec.Type).
+		WithServicePort(&corev1.ServicePort{
+			Name:       "pgbouncer",
+			Port:       pgBouncerConfig.PgBouncerPort,
+			TargetPort: intstr.FromString("pgbouncer"),
+			Protocol:   corev1.ProtocolTCP,
+		}).
+		WithSelector(true, pooler.Name).
```

Comment:
Please invert the two parameters to have the `overwrite` one last.
```suggestion
		WithSelector(pooler.Name, true).
```